Repositories/DbMessageRepository.cs

```diff
@@ -0,0 +1,85 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Karma.Database;
+using Karma.Models;
+using Microsoft.EntityFrameworkCore;
+
+namespace Karma.Repositories
+{
+    public class DbMessageRepository : IMessageRepository
+    {
+        public readonly BaseDbContext _context;
+
+        public DbSet<Message> entities { get; set; }
+
+        public DbMessageRepository(BaseDbContext context)
+        {
+            _context = context;
+            entities = _context.Messages;
+        }
+
+        public IEnumerable<Message> GetAllByGroup(string groupId)
+        {
+            var messages =  from m in entities
+                            where m.GroupId == groupId
+                            select m;
+            return messages;
+        }
+
+        public IEnumerable<Conversation> GetConversations(int userId)
+        {
+            var conversations = from c in _context.Conversations
+                                where c.UserOneId == userId || c.UserTwoId == userId
+                                select c; 
+            return conversations;
+        }
+
+        public Conversation GetConversation(string groupId)
+        {
+            var conversation =  from c in _context.Conversations
+                                where c.GroupId == groupId
+                                select c;
+            return conversation.First();
```

Comment:
.First() throws exception when list is empty, you should probably catch them or do FirstOrDefault and null check where this method is used.

Reply:
Changed to .FirstOrDefault() and added a null check in the corresponding controllers.